src/pages/Profile.js

```diff
@@ -98,40 +138,83 @@ export default function Profile(props) {
+            processPosts(nextPage.value, initLocal);
           }
         }
       };
       window.addEventListener("scroll", loadNext);
     }
-  }, [postLoader, allLoaded]);
+  }, [postLoader, allLoaded, initLocal]);
 
   //execute this effect on entry and when the feedDAC connection status is valid
   useEffect(() => {
-    //setLoading(true);
-    //setProfile(null);
     if (!isMySkyLoading && feedDAC.connector) {
+      //get feed using only remote means
+      const getInitFeed = async (usedLocal = false) => {
+        const postsLoader = await loadBlogProfile(
+          id.substring(8),
+          feedDAC,
+          client
+        );
+        const page0 = await postsLoader.next();
+        setPostLoader(postsLoader);
+        console.log("FEED: ", page0);
+        //loop and variables for loading in resolver links in parallel
+        if (page0.done) setAllLoaded(true);
+        let postArr = page0.value;
+        processPosts(postArr, usedLocal);
+      };
+      //if it is current user's profile, use local storage
+      const getMyInitFeed = async () => {
+        const localFeed = JSON.parse(localStorage.getItem("myFeed"));
+        if (localFeed) {
+          setUpdating(true);
+          setInitLocal(true);
+          let dataFeed = [];
+          localFeed.forEach((item) => {
+            const localPost = JSON.parse(localStorage.getItem(item));
+            if (localPost.isPinned) setPinnedPosts(true);
+            dataFeed.push(localPost);
+          });
+          setPostFeed(dataFeed);
+          setLoading(false);
+          setMoreLoading(false);
+          getInitFeed(true);
+        } else {
+          getInitFeed();
+        }
+      };
+
       //handle retrieval of profile DAC data and feed array
       const getProfileData = async () => {
-        setShowError(false);
-        const profile = await getUserProfile(id.substring(8));
         setMine(id.substring(8) === userID);
-        if (!profile.error) {
-          setProfile(profile);
-          const postsLoader = await loadBlogProfile(
-            id.substring(8),
-            feedDAC,
-            client
-          );
-          const page0 = await postsLoader.next();
-          setPostLoader(postsLoader);
-          console.log("FEED: ", page0);
-          //loop and variables for loading in resolver links in parallel
-          if (page0.done) setAllLoaded(true);
-          let postArr = page0.value;
-          processPosts(postArr);
+        setShowError(false);
+        const prof = JSON.parse(localStorage.getItem(id.substring(8)));
+        if (prof) {
+          setProfile(prof);
+          if (id.substring(8) === userID) {
+            getMyInitFeed();
+          } else {
+            getInitFeed();
+          }
+          const profile = await getUserProfile(id.substring(8));
+          if (!isEqual(profile, prof)) {
+            setProfile(profile);
+            localStorage.setItem(id.substring(8), JSON.stringify(profile));
```

Comment:
I would then make methods for each item, like `setLocalStorageProfile` that accepts the profile. You wouldn't even need to pass in the key, since the key could just be referenced in the method to reduce errors of passing in the wrong key.